alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt

```diff
@@ -234,6 +296,55 @@ class SecureMonitorRestApiIT : AlertingRestTestCase() {
         assertUserNull(createResponse.asMap()["monitor"] as HashMap<String, Any>)
     }
 
+    fun `test get monitor with an user with get monitor role`() {
+        createUserWithTestDataAndCustomRole(
+            user,
+            TEST_HR_INDEX,
+            TEST_HR_ROLE,
+            TEST_HR_BACKEND_ROLE,
+            getClusterPermissionsFromCustomRole(ALERTING_GET_MONITOR_ACCESS)
+        )
+
+        val monitor = createRandomMonitor(true)
+
+        try {
+            val getMonitorResponse = userClient?.makeRequest(
+                "GET",
+                "$ALERTING_BASE_URI/${monitor.id}",
+                null,
+                BasicHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+            )
+            assertEquals("Get monitor failed", RestStatus.OK, getMonitorResponse?.restStatus())
```

Comment:
Could this assertion get missed if there occurs an exception during the request execution. See if catch bock is useful here. Applicable at multiple places.